### ISSUE TITLE
Calendar - range of dates color accessibility

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7364]
The Calendar dates within a range are indicated by a light purple color. This range background color on the dates is the only visual communication given that the dates are selected as part of a range. This color doesn't meet accessibility contrast requirements. See https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html.

Someone with a vision disability may have a hard time distinguishing a range of dates vs two dates being selected:

<img width="417" alt="Screenshot 2024-10-14 at 5 12 08 PM" src="https://github.com/user-attachments/assets/230f2a22-cb1d-46da-bbd6-8672b6ba2d74">

<img width="440" alt="Screenshot 2024-10-14 at 5 13 10 PM" src="https://github.com/user-attachments/assets/c36b61aa-2905-4323-9ac6-4ab3dccbf382">

<img width="693" alt="Screenshot 2024-10-14 at 5 19 36 PM" src="https://github.com/user-attachments/assets/65550d16-e56c-4116-b975-33219dd9348d">

We should explore other visual options for communicating that dates are part of a range.

**Comment by SaunakNandi** [comment]
So can we increase the color contrast to any hex value? orr is there any specific color requirement? @grommetux and @jcfilben

**Comment by jcfilben** [comment]
Instead of changing the color we are changing the font weight to give another visual indicator that the dates are part of a range. See #7402
